### ISSUE TITLE
Null values are getting written when using a custom put resolver to prevent it

### ISSUE THREAD

**Issue by sreejithraman** [pushtorefresh/storio#736]
I created a put resolver that looks like this: 
````java
public class UserPutResolver extends UserStorIOSQLitePutResolver {

  @NonNull
  @Override
  public ContentValues mapToContentValues(@NonNull User user) {
    ContentValues contentValues = super.mapToContentValues(user);
    if(TextUtils.isEmpty(user.getEmail())) {
      contentValues.remove(UsersTable.COLUMN_EMAIL);
    }
    return contentValues;
  }
}
````

Bascially if the user's email is null, I don't want to override it, so I remove the content value. When debugging, I know it hits this line and properly returns the ContentValues object without the email column.

I know I can use the `ignorenull` property, but i want to know why this is not working.

**Comment by sreejithraman** [comment]
So, debugging more, [I fixed it but I have no clue why it works.](http://images-cdn.9gag.com/photo/anYZ9Eo_700b.jpg)
In the code above, I left out another thing I was doing. Here is what it actually looked like:
````java
public class UserPutResolver extends UserStorIOSQLitePutResolver {

  @NonNull
  @Override
  public ContentValues mapToContentValues(@NonNull User user) {
    ContentValues contentValues = super.mapToContentValues(user);
    if(TextUtils.isEmpty(user.getEmail())) {
      contentValues.remove(UsersTable.COLUMN_EMAIL);
    }
    if(user.getFriendshipStatus() == null){
      contentValues.remove(UsersTable.COLUMN_EMAIL);
    }

    return contentValues;
  }
}
````
Notice that I was removing the email column twice. When I fixed it to remove the correct column, it magically started working. Does anyone know why this would be the case??

**Comment by nikitin-da** [comment]
Hi, @sreejithraman !
It is strange that duplicate remove value from map make it to work=)
I can't reproduce you case, can you please help me on branch (https://github.com/pushtorefresh/storio/commit/675bb6216762acc7c62730f08fcbed9a283564f8#diff-24675be996f2e24c996d51fdcedbcbf7R32) or give some another example?

**Comment by sreejithraman** [comment]
@nikitin-da no, it wasn't adding the duplicate remove, it was removing the duplicate remove. I had initailly taken out extra code in my original post thinking the issue was in that code block. Later when I realized that the issue was that I was calling `contentValues.remove(UsersTable.COLUMN_EMAIL);` twice, I posted all of the code I originally had to show what was happening. Notice that in my if statement, i meant to remove the Friendship column, but i was removing the email column again by mistake. Changing the duplicate `contentValues.remove(UsersTable.COLUMN_EMAIL);` to `contentValues.remove(UsersTable.COLUMN_FRIEND_SHIP_STATUS);`  fixed the issue.

Sorry for the confusion. I commented on that branch you linked. Lemme know if that produces the bug.

**Comment by nikitin-da** [comment]
I don't think that the cause may be in the duplicate removing) 
I added the second call as you advised, in branch (https://github.com/pushtorefresh/storio/commit/0a9906a09c28262ca00b491c1a49f84636daffc4#diff-24675be996f2e24c996d51fdcedbcbf7R51) and test was passed. 
Can you please pull this branch and make changes to break it or share some simple example where you can reproduce it ;)

**Comment by sreejithraman** [comment]
hmm.. interesting. yea, i'll try to reproduce it and let you know if I am successful
